### PR TITLE
Move version bump logic into publish task

### PR DESCRIPTION
**Pull request by Colin1860**
Since bumping versions is only really necessary when publishing the affected logic is moved into the publish task and the dedicated bump-task removed

### CODE REVIEW

File: kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py [hunk 1]
```diff
@@ -53,3 +65,49 @@ def make_safe(self, args: list[str], env: dict[str, str]) -> None:
...
+    def execute(self) -> TaskStatus:
+        version = self.version.get()
+        if version is not None:
+            content = self._get_updated_cargo_toml(version)
+            with atomic_file_swap(self.cargo_toml_file.get(), "w", always_revert=True) as ctx:
+                ctx.write(content)
+                ctx.close()
+                return super().execute()
+        else:
+            return super().execute()
```
Comment (NiklasRosenstein) [inline kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py]:
```suggestion
        with contextlib.ExitStack() as stack:
            if (version := self.version.get()) is not None:
                fp = stack.enter_context(atomic_file_swap(self.cargo_toml_file.get(), "w", always_revert=True))
                fp.write(content)
                fp.close()
            return super().execute()
```

I think something like this is more concise

**Review by NiklasRosenstein — COMMENTED** [review]
Looks good to me overall! :)

File: kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py (hunk 1 above)
Comment (Colin1860) [reply kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py]: I've had something similar like this but then, python complains that all branches of execution must return a value, and if i move the return statement outside of the exitstack scope, the file is reverted before super().execute()

Comment (NiklasRosenstein) [reply kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py]: Ah, yes this is a bit of an unfortunate quirk with the `with` statement and type checking imo. Using

```py
    result = super().execute()
return result
```

should work?

File: kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py [hunk 2]
```diff
@@ -53,3 +65,48 @@ def make_safe(self, args: list[str], env: dict[str, str]) -> None:
...
+                    dependency["registry"] = registry_alias
+
+    def execute(self) -> TaskStatus:
+        with contextlib.ExitStack() as stack:
+            if (version := self.version.get()) is not None:
+                content = self._get_updated_cargo_toml(version)
+                fp = stack.enter_context(atomic_file_swap(self.cargo_toml_file.get(), "w", always_revert=True))
+                fp.write(content)
+                fp.close()
+                result = super().execute()
```
Comment (NiklasRosenstein) [inline kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py]: This has one too many indent. We want to run super.execute regardless of whether a version is set to bump or not.

**Review by NiklasRosenstein — CHANGES_REQUESTED** [review]

File: kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py (hunk 2 above)
Comment (Colin1860) [reply kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py]: Alright, that's what i thought. But then i won't get around writing `super().execute()` twice i think

Comment (Colin1860) [reply kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py]: Or this?
```
result = TaskStatus.pending()
with contextlib.ExitStack() as stack:
    if (version := self.version.get()) is not None:
        content = self._get_updated_cargo_toml(version)
        fp = stack.enter_context(atomic_file_swap(self.cargo_toml_file.get(), "w", always_revert=True))
        fp.write(content)
        fp.close()
    result = super().execute()

return result
```

Comment (Colin1860) [reply kraken-build/src/kraken/std/cargo/tasks/cargo_publish_task.py]: Ok, now i understand. Didn't know this was possible without first initializing the result variable.

**Pull request merged**